### ISSUE TITLE
TAU_MAKEFILE env variable

### ISSUE THREAD

**Issue by brnorris03** [HPCL/autoperf#14]
Users shouldn't have to set any env. variables when running autoperf, right now instrumentation-based profiling is asking for TAU_MAKEFILE. This should be set in the config file instead.

**\* Preparing to run pi_inst
make: Entering directory `/home/norris/temp/autoperf1'
tau_cc.sh -Wall pi.c -o pi_inst
/home/norris/soft/tau2/x86_64/bin/tau_cc.sh: ERROR: please set the environment variable TAU_MAKEFILE
make: **\* [pi_inst] Error 1

**Comment by brnorris03** [comment]
Also I don't think this should be requiring the full path because it is already specified in tauroot

**Comment by xdai** [comment]
Autoperf depends on user script to build the user application, instrumented or not. I prefer to leave all the build related details out of autoperf. What do you think?

**Comment by brnorris03** [comment]
I don't think we can avoid having to deal with some of this if autoperf supports instrumentation (which it does now). I think the tool configurations need to be handled better in general, but this particular problem is easily fixed by adding another tau option.

**Comment by xdai** [comment]
How about a generic `[Env]` section and export all options under it as envrionment variables to all the sub-processes?

**Comment by brnorris03** [comment]
That would work, but it brings up a different issue. I think the configuration options should be split into general (all machines) and local (current machine only). It is very messy and error-prone to have multiple almost-identical config files. Assuming one may want to set up experiments on multiple machines, it may be good to also help organize them, i.e., putting results in separate subdirectories and including the machine name in the experiment names in the database. This may have to be a different issues because it's a more significant change.
